src/main/relaxng/docbook/programming.rnc

```diff
@@ -112,7 +112,7 @@ div {
    db.synopsisinfo =
       element synopsisinfo {
          db.synopsisinfo.attlist,
-        (db.verbatim.contentmodel | (db.programming.inlines?, db.all.blocks*))
+         db.verbatim.contentmodel
```

Comment:
You can put `xlink:href` on any inline that you put in the verbatim environment. But a `synopsisinfo` has to remain a verbatim environment, it can't be ordinary blocks like `para`. That would be a huge change and would violate the processing expectations that have been in place for decades.

Reply:
How would you then represent pure metadata about a function (i.e. something that does not have to be output to the user, depending on the context)? For instance, to indicate that a given method overrides another: there is not always appropriate syntax in the programming language to indicate this, it's only relevant in documentation (and actual execution, but that's off-topic). 

I think it's still better to give it inside the `methodsynopsis`, to continue that example, so that the processor may decide to generate a link at the appropriate place (for instance, with some generated text), and to generate a back link from the overridden method (which would be close to impossible to implement if the link to the overridden method is only available in a `para` "near" the `methodsynopsis`). You may also want to generate a version of your manual without too much information for beginners, which would mean without links to overridden methods (because that would just clutter the display). 

Just to show what that text generation could do: 

```xml
<section>
  <!-- … -->
  <methodsynopsis xml:id="method.overriddes">
    <synopsisinfo role="overrides"><link linkend="method.overridden">method.overridden(…)</link></synopsisinfo>
    <!-- … -->
  </methodsynopsis>
  <!-- … -->
</section>
<section>
  <!-- … -->
  <methodsynopsis xml:id="method.overridden">
    <!-- … -->
  </methodsynopsis>
  <!-- … -->
</section>
```

```xml
<section>
  <!-- … -->
  <para>This method overrides <link linkend="method.overridden">method.overridden(…)</link>.</para>
  <methodsynopsis xml:id="method.overriddes">
    <synopsisinfo role="overrides"><link linkend="method.overridden">method.overridden(…)</link></synopsisinfo>
    <!-- … -->
  </methodsynopsis>
  <!-- … -->
</section>
<section>
  <!-- … -->
  <para>This method is overridden by: <link linkend="method.overriddes">method.overriddes(…)</link>.</para>
  <methodsynopsis xml:id="method.overridden">
    <!-- … -->
  </methodsynopsis>
  <!-- … -->
</section>
```